server/controllers/dog.js

```diff
@@ -0,0 +1,128 @@
+const User = require("../models/User");
+const Dog = require("../models/Dog");
+const asyncHandler = require("express-async-handler");
+const ObjectId = require("mongoose").Types.ObjectId;
+
+// @route GET /dogs/:id:
+// @desc Get dog for logged in user by :id
+// @access Private
+exports.loadDog = asyncHandler(async (req, res, next) => {
+  const id = req.params.id;
+  const dog = await Dog.findById(id);
+
+  if (dog) {
+    res.status(200).send({
+      success: {
+        dog,
+      },
+    });
+  } else res.status(400).json({ errors: "Bad request !" });
+});
+
+// @route GET /dogs
+// @desc Get dogs for logged in user
+// @access Private
+exports.listDogs = asyncHandler(async (req, res, next) => {
+  const user = await User.findById(req.user.id);
+  if (!user) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+
+  const dogs = await Dog.find({ userId: ObjectId(user._id) });
+  res.status(200).json({
+    success: {
+      dogs,
+    },
+  });
+});
+
+// @route POST /dogs
+// @desc create new dog for logged in user (dog owner)
+// @access Private
+exports.createDog = asyncHandler(async (req, res, next) => {
+  const newDog = ({
```

Comment:
Good work destructing the request body here 👍🏽

Reply:
thank you :)